src/main/java/seedu/address/logic/parser/RecipeBookParser.java

```diff
@@ -41,8 +43,18 @@ public Command parseCommand(String userInput) throws ParseException {
 
         final String commandWord = matcher.group("commandWord");
         final String arguments = matcher.group("arguments");
-        switch (commandWord) {
 
+        /* If previous command is a clear command, enforceConfirmation will ensure the next command
+        a confirmationCommand. */
+        if (EnforceConfirmation.clearIsCalled()) {
+            boolean isConfirmation = EnforceConfirmation.isNextCommandAConfirmation(commandWord);
+            if (!isConfirmation) {
+                EnforceConfirmation.resetConfirmation();
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                        ConfirmationCommand.ENFORCE_MESSAGE));
+            }
+        }
```

Comment:
This block makes sense but feels a little out of place, but I'm not quite sure where else to put this if not here.